### DISCUSSION TITLE
Keyboard Interrupt Error When Running Dev Instance of Uvicorn

### DISCUSSION THREAD

**Discussion by pollyandhiscat** [fastapi/fastapi#11643] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Union

from fastapi import FastAPI

app = FastAPI()

@app.get("/")
def read_root():
    return {"Greetings": "Earthling"}
```

### Description

When you have Uvicorn running in dev mode with FastAPI and save the 'main.py' file while the program is running, a quite monstrous error appears in the terminal, which ultimately amounts to a keyboard interrupt error. The following screenshots illustrate the issue:

![image](https://github.com/tiangolo/fastapi/assets/143892700/5b9958c0-1ab9-45e6-8b3f-bc7f7d9dcb04)

![image](https://github.com/tiangolo/fastapi/assets/143892700/4338563e-b5e2-4d1d-a196-b3c670c6c2b2)

![image](https://github.com/tiangolo/fastapi/assets/143892700/fb6ed40b-ab35-4606-8ec0-c974113d385a)

![image](https://github.com/tiangolo/fastapi/assets/143892700/ed8166a4-dfa7-463d-8d83-ae3b4e74bf0d)

### Operating System

Windows

### Operating System Details

Windows 11 Pro

### FastAPI Version

0.111.0

### Pydantic Version

2.7.0

### Python Version

3.12.2

**Comment by YuriiMotov** [comment]
Where were some problems in Uvicorn that caused such errors.
I believe it's been fixed.
